core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -453,19 +456,85 @@ public void writeOptionalStreamable(@Nullable Streamable streamable) throws IOEx
         }
     }
 
+    // Remove after upgrade to lucene 5.3
```

Comment:
Can you add a static assertion that will fail when we upgrade to Lucene 5.3 like we do for X-classes?